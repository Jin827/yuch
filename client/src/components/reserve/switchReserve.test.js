import React from 'react';
import { shallow } from 'enzyme';
import SwitchReserve from './switchReserve';

describe('<switchReserve />', () => {
  const props = {
    apiRequest: null,
  };
  const wrapper = shallow(<SwitchReserve {...props} />);

  it('render reserve form when api request not made', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('ReserveForm').exists()).toBeTruthy();
  });

  it('render reserve request resolved text when api request is made', () => {
    wrapper.setProps({ apiRequest: 'success' });
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper.find('WithStyles(ReserveResolvedText)').exists(),
    ).toBeTruthy();
  });
});
