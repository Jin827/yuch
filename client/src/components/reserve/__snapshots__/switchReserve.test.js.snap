// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<switchReserve /> render reserve form when api request not made 1`] = `
<Fragment>
  <h3
    className="f-xl"
    id="modal-title"
    variant="title"
  >
    - Reservation -
  </h3>
  <ReserveForm />
</Fragment>
`;

exports[`<switchReserve /> render reserve request resolved text when api request is made 1`] = `
<Fragment>
  <h3
    className="f-xl"
    id="modal-title"
    variant="title"
  >
    - Reservation -
  </h3>
  <WithStyles(ReserveResolvedText)
    apiRequest="success"
  />
</Fragment>
`;
